app/helpers/application_helper.rb

```diff
@@ -2,4 +2,43 @@
 
 # Any needed Application UI functions
 module ApplicationHelper
+  # Display buttons for the show page
+  # Only buttons that the user has permission to use are shown.
+  def buttons_for_show(resource)
+    buttons = []
+    buttons << edit_link(resource)
+    buttons << delete_link(resource)
+    buttons << index_link(resource)
+    safe_join(buttons.compact_blank, "|")
+  end
+
+  # Display links on Index
```

Comment:
Do we need 
> # Only buttons that the user has permission to use are shown.

To make the comment consistent with the one for `buttons_for_show`

Reply:
Same for the other helpers.

Reply:
I rewrote the comments here, because we move permissions to other helper methods.